src/pages/SkillsEditor/SkillEditorList.spec.tsx

```diff
@@ -0,0 +1,135 @@
+import { render, waitFor, screen, fireEvent } from "@testing-library/react";
+import { createMemoryHistory } from "history";
+import { Router } from "react-router-dom";
+import React, { FunctionComponent } from "react";
+import { createTheme, ThemeProvider } from "@mui/material";
+import { mocked } from "jest-mock";
+import { useFirebaseApp } from "../../context/FirebaseContext/FirebaseContext";
+import { useAppState } from "../../context/AppStateContext/AppStateContext";
+import { useSkillsContext } from "../../context/SkillsContext/SkillsContext";
+import SkillsEditorList from "./SkillsEditorList";
+
+const ThemeProviderWrapper: FunctionComponent = ({ children }) => {
+  const theme = createTheme({});
+  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
+};
+
+jest.mock("../../context/FirebaseContext/FirebaseContext");
+jest.mock("../../context/SkillsContext/SkillsContext");
+jest.mock("../../context/AppStateContext/AppStateContext");
+
+describe("Skill List", () => {
+  const history = createMemoryHistory();
+
+  beforeEach(() => {
+    jest.resetAllMocks();
+
+    mocked(useSkillsContext).mockReturnValue({
+      skillList: ["JS"],
+      updateSkillList: jest.fn(),
+    });
+
+    mocked(useAppState).mockReturnValue({
+      isDrawerOpen: false,
+      setIsDrawerOpen: jest.fn(),
+    });
+
+    mocked(useFirebaseApp).mockImplementation(
+      () =>
+        ({
+          firebase: {},
+        } as any)
+    );
+  });
+
+  it("Should render page and change new skill name", async () => {
+    render(
+      <Router history={history}>
+        <ThemeProviderWrapper>
+          <SkillsEditorList />
+        </ThemeProviderWrapper>
+      </Router>
+    );
+
+    await waitFor(() => {
+      const input = screen.getByPlaceholderText("Skill name") as HTMLInputElement;
+      expect(input).toBeInTheDocument();
+      fireEvent.input(input, { target: { value: "new" } });
+      expect(input.value).toBe("new");
+
+      const form = screen.getByTestId("form");
+      fireEvent.submit(form);
+    });
```

Comment:
I think it would be better to only put this function call in `waitFor` and keep the rest of the logic outside of waitFor. That way we only wait until necessary conditions are there and then continue with our test.

What I have in mind is something like this:

```suggestion
    await waitFor(() => expect(screen.getByPlaceholderText("Skill name")).toBeInTheDocument());

    const input = screen.getByPlaceholderText("Skill name") as HTMLInputElement;
    fireEvent.input(input, { target: { value: "new" } });
    expect(input.value).toBe("new");

    const form = screen.getByTestId("form");
    fireEvent.submit(form);
```